internal/wizard/project/create.go

```diff
@@ -0,0 +1,291 @@
+						// let's sort orgs
+						sort.Slice(organizations, func(i, j int) bool {
+							return organizations[i].Name < organizations[j].Name
+						})
+						options := make([]huh.Option[string], len(organizations))
+						for i, org := range organizations {
+							orgProjectCount := len(org.Projects)
+							if orgProjectCount >= org.QuotaProject && org.QuotaProject >= 0 {
+								quotaFullLabel := lipgloss.NewStyle().Foreground(lipgloss.Color("240")).Render(fmt.Sprintf("%s | Project Limit: %v/%v | ⚠️  Project Quota full, cannot assigned project to this organization.", org.Name, orgProjectCount, org.QuotaProject))
+								options[i] = huh.NewOption(quotaFullLabel, org.Name)
+							} else {
+								options[i] = huh.NewOption(fmt.Sprintf("%s | Project Limit: %v/%v", org.Name, orgProjectCount, util.QuotaCheck(org.QuotaProject)), org.Name)
+							}
+						}
+						return options
+					}, &config.OrganizationDetails.Name).
+					Value(&config.OrganizationDetails.Name).
+					Validate(func(selectedOrg string) error {
+						for _, org := range organizations {
+							if org.Name == selectedOrg {
+								orgProjectCount := len(org.Projects)
+								if orgProjectCount >= org.QuotaProject && org.QuotaProject >= 0 {
+									orgValidationErr = true
+									return nil
+								}
+							}
+						}
+						return nil
+					}),
+			),
+		).WithTheme(themes.ThemeLagoon())
+
+		err = form2.Run()
+		if err != nil {
+			return nil, err
+		}
+
+		if orgValidationErr {
+			var orgRetryResp string
+			orgRetryRespForm := huh.NewForm(
+				huh.NewGroup(
+					huh.NewSelect[string]().
+						Title("Organization quota exceeded. What would you like to do?").
+						Options(
+							huh.NewOption("Go back to previous form", "back"),
+							huh.NewOption("Cancel wizard", "cancel"),
+						).
+						Value(&orgRetryResp),
+				),
+			).WithTheme(themes.ThemeLagoon())
+
+			err := orgRetryRespForm.Run()
+			if err != nil {
+				return nil, err
+			}
+
+			switch orgRetryResp {
+			case "back":
+				return RunCreateWizard(lc)
+			case "cancel":
+				return nil, huh.ErrUserAborted
+			}
+		}
+	}
+
+	var deployTargets *[]schema.DeployTarget
+	var err error
+	if organizationConfirm {
+		orgDeploytargets, err := util.GetOrgDeployTargets(lc, config.OrganizationDetails.Name)
+		if err != nil {
+			return nil, err
+		}
+		deployTargets = &orgDeploytargets
+	} else {
+		deployTargets, err = lagoon.ListDeployTargets(context.TODO(), lc)
+		if err != nil {
+			return nil, err
+		}
+		if deployTargets == nil {
+			return nil, errors.New("no deployTargets found for user")
+		}
+	}
+	var additionalFields bool
+	options := make([]huh.Option[uint], len(*deployTargets))
+	for i, target := range *deployTargets {
+		options[i] = huh.NewOption(target.Name, target.ID)
+	}
+	form3 := huh.NewForm(
+		huh.NewGroup(
+			huh.NewInput().
+				Title("Enter the Git URL for your project").
+				Value(&config.Input.GitURL).
+				Validate(func(str string) error {
```

Comment:
We do giturl validation in the API https://github.com/uselagoon/lagoon/blob/00c6a609bd787b7d451ee8371a547eab6d5bf445/services/api/src/resources/project/resolvers.ts#L294

Do we need to also validate here? I think this validation method is slightly different to the one in the API too, meaning some things work in the API, but are rejected by this validator.
eg:
API accepts `https://git.com/repo/example` while this validator rejects.

I think omit, and rely on the API to validate, or reproduce https://www.npmjs.com/package/git-url-parse in go :see_no_evil:

Reply:
Ideally the validation remains here from a UX perspective. Initially it was being handled by the api, but the response from the api that the url is invalid is only being provided on completion of this wizard, forcing the user to have to re-launch and re-enter everything just to correct the giturl.

With it in the form the user can't progress until that giturl is actually valid. I've also made the url validation more robust to be closer to git-url-parse, so hopefully that should fix any inconsistencies with validation here vs the api :crossed_fingers:. But totally open to other options